cmd/celestia-da/main.go

```diff
@@ -78,13 +44,104 @@ func serve(rpcAddress string, rpcToken string, nsString string) {
 	// TODO(tzdybal): add configuration options for encryption
 	srv := proxy.NewServer(da, grpc.Creds(insecure.NewCredentials()))
 
-	lis, err := net.Listen("tcp", "")
+	lis, err := net.Listen(listenNetwork, listenAddress)
 	if err != nil {
 		log.Fatalln("failed to create network listener:", err)
 	}
-	log.Println("serving celestia-da over gRPC on:", lis.Addr())
+	defer func() {
+		if err := lis.Close(); err != nil {
+			log.Errorln("failed to close network listener:", err)
+		}
+	}()
+	log.Infoln("serving celestia-da over gRPC on:", lis.Addr())
 	err = srv.Serve(lis)
 	if !errors.Is(err, grpc.ErrServerStopped) {
 		log.Fatalln("gRPC server stopped with error:", err)
 	}
 }
+
+// WithDataAvailabilityService patches the start command to also run the gRPC Data Availability service
+func WithDataAvailabilityService(flags []*pflag.FlagSet) func(*cobra.Command) {
+	return func(c *cobra.Command) {
+		grpcFlags := &pflag.FlagSet{}
+		grpcFlags.String("grpc.address", "http://127.0.0.1:26658", "celestia-node RPC endpoint address")
+		grpcFlags.String("grpc.token", "", "celestia-node RPC auth token")
+		grpcFlags.String("grpc.namespace", "", "celestia namespace to use (hex encoded)")
+		grpcFlags.String("grpc.listen", "", "gRPC service listen address")
+		grpcFlags.String("grpc.network", "tcp", "gRPC service listen network type must be \"tcp\", \"tcp4\", \"tcp6\", \"unix\" or \"unixpacket\"")
+
+		fset := append(flags, grpcFlags)
+
+		for _, set := range fset {
+			c.Flags().AddFlagSet(set)
+		}
+
+		requiredFlags := []string{"grpc.token", "grpc.namespace"}
+		for _, required := range requiredFlags {
+			if err := c.MarkFlagRequired(required); err != nil {
+				log.Fatal(required, err)
+			}
+		}
+
+		preRun := func(cmd *cobra.Command, args []string) {
+			// Extract gRPC service flags
+			rpcAddress, _ := cmd.Flags().GetString("grpc.address")
+			rpcToken, _ := cmd.Flags().GetString("grpc.token")
+			nsString, _ := cmd.Flags().GetString("grpc.namespace")
+			listenAddress, _ := cmd.Flags().GetString("grpc.listen")
+			listenNetwork, _ := cmd.Flags().GetString("grpc.network")
```

Comment:
nit: Introduce constants for flag names.

Reply:
Done 👍🏽

Reply:
Also want to prefix these with "da" to avoid confusion.